File: native/Travel Companion/Model/Google/GooglePlaceType.swift
//
//  GooglePlaceType.swift
//  Travel Companion
//
//  Created by Stefan Jaindl on 31.10.18.
//  Copyright © 2018 Stefan Jaindl. All rights reserved.
//

import Foundation

public enum GooglePlaceType: String {
    case lodging           //hotels
    case restaurant        //restaurants
    
    //attractions:
    case pointOfInterest
    case amusementPark
    case aquarium
    case artGallery
    case atm
    case bank
    case bar
    case beautySalon
    case bowlingAlley
    case cafe
    case casino
    case church
    case cityHall
    case embassy
    case gym
    case hinduTemple
    case library
    case mosque
    case movieTheater
    case museum
    case nightClub
    case postOffice
    case rvPark
    case shoppingMall
    case spa
    case stadium
    case synagogue
    case travelAgency
    case zoo
    
    public var key: String {
        switch self {
        case .pointOfInterest:
            return "point_of_interest"
        case .amusementPark:
            return "amusement_park"
        case .artGallery:
            return "art_gallery"
        case .beautySalon:
            return "beauty_salon"
        case .bowlingAlley:
            return "bowling_alley"
        case .cityHall:
            return "city_hall"
        case .hinduTemple:
            return "hindu_temple"
        case .movieTheater:
            return "movie_theater"
        case .nightClub:
            return "night_club"
        case .postOffice:
            return "post_office"
        case .rvPark:
            return "rv_park"
        case .shoppingMall:
            return "shopping_mall"
        case .travelAgency:
            return "travel_agency"
        default:
            return self.rawValue
        }
    }
    
    public var description: String? {
        switch self {
        case .lodging:
            return "lodging".localized()
        case .restaurant:
            return "restaurant".localized()
        case .pointOfInterest:
            return "pointOfInterest".localized()
        case .amusementPark:
            return "amusementPark".localized()
        case .aquarium:
            return "aquarium".localized()
        case .artGallery:
            return "artGallery".localized()
        case .atm:
            return "atm".localized()
        case .bank:
            return "bank".localized()
        case .bar:
            return "bar".localized()
        case .beautySalon:
            return "beautySalon".localized()
        case .bowlingAlley:
            return "bowlingAlley".localized()
        case .cafe:
            return "cafe".localized()
        case .casino:
            return "casino".localized()
        case .church:
            return "church".localized()
        case .cityHall:
            return "cityHall".localized()
        case .embassy:
            return "embassy".localized()
        case .gym:
            return "gym".localized()
        case .hinduTemple:
            return "hinduTemple".localized()
        case .library:
            return "library".localized()
        case .mosque:
            return "mosque".localized()
        case .movieTheater:
            return "movieTheater".localized()
        case .museum:
            return "museum".localized()
        case .nightClub:
            return "nightClub".localized()
        case .postOffice:
            return "postOffice".localized()
        case .rvPark:
            return "rvPark".localized()
        case .shoppingMall:
            return "shoppingMall".localized()
        case .spa:
            return "spa".localized()
        case .stadium:
            return "stadium".localized()
        case .synagogue:
            return "synagogue".localized()
        case .travelAgency:
            return "travelAgency".localized()
        case .zoo:
            return "zoo".localized()
        }
    }
}

File: shared/src/commonMain/kotlin/com.sjaindl.travelcompanion/api/google/GooglePlaceType.kt
package com.sjaindl.travelcompanion.api.google

//
//  GooglePlaceType.swift
//  Travel Companion
//
//  Created by Stefan Jaindl on 31.10.18.
//  Copyright © 2018 Stefan Jaindl. All rights reserved.
//
enum class GooglePlaceType(private val rawValue: String) {
    Lodging("lodging"),
    Restaurant("restaurant"),
    PointOfInterest("pointOfInterest"),
    AmusementPark("amusementPark"),
    Aquarium("aquarium"),
    ArtGallery("artGallery"),
    Atm("atm"),
    Bank("bank"),
    Bar("bar"),
    BeautySalon("beautySalon"),
    BowlingAlley("bowlingAlley"),
    Cafe("cafe"),
    Casino("casino"),
    Church("church"),
    CityHall("cityHall"),
    Embassy("embassy"),
    Gym("gym"),
    HinduTemple("hinduTemple"),
    Library("library"),
    Mosque("mosque"),
    MovieTheater("movieTheater"),
    Museum("museum"),
    NightClub("nightClub"),
    PostOffice("postOffice"),
    RvPark("rvPark"),
    ShoppingMall("shoppingMall"),
    Spa("spa"),
    Stadium("stadium"),
    Synagogue("synagogue"),
    TravelAgency("travelAgency"),
    Zoo("zoo");

    //hotels
    //restaurants
    //attractions:
    val key: String
        get() {
            when (this) {
                PointOfInterest -> return "point_of_interest"
                AmusementPark -> return "amusement_park"
                ArtGallery -> return "art_gallery"
                BeautySalon -> return "beauty_salon"
                BowlingAlley -> return "bowling_alley"
                CityHall -> return "city_hall"
                HinduTemple -> return "hindu_temple"
                MovieTheater -> return "movie_theater"
                NightClub -> return "night_club"
                PostOffice -> return "post_office"
                RvPark -> return "rv_park"
                ShoppingMall -> return "shopping_mall"
                TravelAgency -> return "travel_agency"
                else -> return this.rawValue
            }
        }

    /*

    val description: String?
        get() {
            when (this) {
                Lodging -> return R.string.lodging
                Restaurant -> return "restaurant".localized()
                PointOfInterest -> return "pointOfInterest".localized()
                AmusementPark -> return "amusementPark".localized()
                Aquarium -> return "aquarium".localized()
                ArtGallery -> return "artGallery".localized()
                Atm -> return "atm".localized()
                Bank -> return "bank".localized()
                Bar -> return "bar".localized()
                BeautySalon -> return "beautySalon".localized()
                BowlingAlley -> return "bowlingAlley".localized()
                Cafe -> return "cafe".localized()
                Casino -> return "casino".localized()
                Church -> return "church".localized()
                CityHall -> return "cityHall".localized()
                Embassy -> return "embassy".localized()
                Gym -> return "gym".localized()
                HinduTemple -> return "hinduTemple".localized()
                Library -> return "library".localized()
                Mosque -> return "mosque".localized()
                MovieTheater -> return "movieTheater".localized()
                Museum -> return "museum".localized()
                NightClub -> return "nightClub".localized()
                PostOffice -> return "postOffice".localized()
                RvPark -> return "rvPark".localized()
                ShoppingMall -> return "shoppingMall".localized()
                Spa -> return "spa".localized()
                Stadium -> return "stadium".localized()
                Synagogue -> return "synagogue".localized()
                TravelAgency -> return "travelAgency".localized()
                Zoo -> return "zoo".localized()
            }
        }

     */
}
